weasyl/controllers/content.py

```diff
@@ -247,57 +290,55 @@ def submit_journal_get_(request):
 @login_required
 @token_checked
 def submit_journal_post_(request):
-    form = request.web_input(title="", rating="", friends="", members="", content="", tags="")
-
-    tags = searchtag.parse_tags(form.tags)
+    tags = searchtag.parse_tags(request.params.get('tags', ''))
 
     if not define.config_read_bool("allow_submit"):
         raise WeasylError("FeatureDisabled")
 
     if not define.is_vouched_for(request.userid):
         raise WeasylError("vouchRequired")
 
-    rating = ratings.CODE_MAP.get(define.get_int(form.rating))
+    rating = ratings.CODE_MAP.get(define.get_int(request.params.get('rating', '')))
     if not rating:
         raise WeasylError("ratingInvalid")
 
     j = orm.Journal()
-    j.title = form.title
+    j.title = request.params.get('title', '')
     j.rating = rating
-    j.content = form.content
+    j.content = request.params.get('content', '')
     j.submitter_ip_address = request.client_addr
     j.submitter_user_agent_id = get_user_agent_id(ua_string=request.user_agent)
-    journalid = journal.create(request.userid, j, friends_only=form.friends,
+    journalid = journal.create(request.userid, j, request.params.get('friends', ''),
                                tags=tags)
-    raise HTTPSeeOther(location="/journal/%i/%s" % (journalid, slug_for(form.title)))
+    raise HTTPSeeOther(location="/journal/%i/%s" % (journalid, slug_for(j.title)))
 
 
 @login_required
 @token_checked
 @supports_json
 def submit_shout_(request):
-    form = request.web_input(userid="", parentid="", content="", staffnotes="", format="")
+    staffnotes = request.params.get('staffnotes', False)
```

Comment:
The `False` here makes the types a bit confusing. How about (re)moving the `get_int`:

```python
staffnotes = int(request.params.get('staffnotes', 0))